internal/config/config.go

```diff
@@ -208,6 +208,126 @@ func (p *Pools) IsEmpty(pool string) bool {
 	return p.ByName[pool] == nil
 }
 
+// bgpCommunityType holds the type of the BGP community. Values currently supported are legacyBGPCommunity and
+// largeBGPCommunity. In the future, this can be expanded with "extendedBGPCommunity".
+type bgpCommunityType uint
+
+const (
+	legacyBGPCommunity bgpCommunityType = iota
+	largeBGPCommunity
+)
+
+// largeBGPCommunityMarker is the prefix that shall be used to indicate that a given community value is of type large
+// community. If and when extended BGP communities will be supported, the largeBGPCommunityMarker allows us to
+// distinguish between extended and large communities.
+const largeBGPCommunityMarker = "large"
+
+// BGPCommunity holds the internal representation of a BGP community. It currently supports legacy and large BGP
+// communities with plans to expand this to extended communities in the future. A BGPCommunity object is intended to be
+// read only and as such all its methods directly operate on the type itself as opposed to using pointer receivers.
+type BGPCommunity struct {
```

Comment:
This deserves to be (at least) in a separate file, probably even a package (maybe under bgp) as this is accessed from multiple parts.